src/examples/java/io/nats/examples/NatsJsSubCb.java

```diff
@@ -0,0 +1,104 @@
+    static final String usageString = "\nUsage: java NatsJsSub [-s server] <subject> <msgCount>\n"
+            + "\nUse tls:// or opentls:// to require tls, via the Default SSLContext\n"
+            + "\nSet the environment variable NATS_NKEY to use challenge response authentication by setting a file containing your private key.\n"
+            + "\nSet the environment variable NATS_CREDS to use JWT/NKey authentication by setting a file containing your user creds.\n"
+            + "\nUse the URL for user/pass/token authentication.\n";
+
+    public static void main(String[] args) {
+        ExampleArgs exArgs = ExampleUtils.readSubscribeArgs(args, usageString);
+
+        System.out.printf("\nTrying to connect to %s, and listen to %s for %d messages.\n\n", exArgs.server,
+                exArgs.subject, exArgs.msgCount);
+
+        try (Connection nc = Nats.connect(ExampleUtils.createExampleOptions(exArgs.server, true))) {
+
+            // create a dispatcher without a default handler.
+            Dispatcher d = nc.createDispatcher(null);
+
+            // Create our jetstream context to subscribe to jetstream 
+            // messages.
+            JetStream js = nc.jetStream();
+
+            CountDownLatch msgLatch = new CountDownLatch(exArgs.msgCount);
+
+            // create our message handler.
+            MessageHandler handler = (Message msg) -> {
+
+                System.out.printf("\nMessage Received:\n");
+
+                if (msg.hasHeaders()) {
+                    System.out.println("  Headers:");
+                    for (String key: msg.getHeaders().keySet()) {
+                        for (String value : msg.getHeaders().values(key)) {
+                            System.out.printf("    %s: %s\n", key, value);
+                        }
+                    }
+                }
+
+                System.out.printf("  Subject: %s\n  Data: %s\n",
+                        msg.getSubject(),
+                        new String(msg.getData(), StandardCharsets.UTF_8));
+
+                // This check may not be necessary for this example depending
+                // on how the consumer has been setup.  When a deliver subject
+                // is set on a consumer, messages can be received from applications
+                // that are NATS producers and from streams in NATS servers.
+                if (msg.isJetStream()) {
+                    MetaData meta = msg.metaData();
+                    System.out.printf("  Stream Name: %s\n", meta.getStream());
+                    System.out.printf("  Stream Seq:  %d\n",  meta.streamSequence());
+                    System.out.printf("  Consumer Name: %s\n", meta.getConsumer());
+                    System.out.printf("  Consumer Seq:  %d\n", meta.consumerSequence());
+
+                    // If you don't disable auto ack mode NATS will automatically
+                    // ack for you.  Otherwise, you'll need to ack, e.g.
+                    // msg.ack();                    
```

Comment:
make that an option?

Reply:
I'd like to keep this really simple - different ack modes will make this example messy for users to copy, and ack modes need to correspond to consumer configuration.  Documentation with a few new examples and with corresponding configurations will be better imo.  We had a lot of options in Stan, and the example code isn't all that usable imo.